### PR TITLE
Make CI actually work on Windows

### PR DESCRIPTION
**Pull request by CarterLi**

### CODE REVIEW

**Comment by LinusDierheimer** [comment]
CI seems fine, and i can't really judge the windows code. Can i merge this now or are you still working on it?

File: src/detection/os/os_windows.cpp [hunk 1]
```diff
@@ -33,13 +33,21 @@ void ffDetectOSImpl(FFOSResult* os, const FFinstance* instance)
...
 
-        ffStrbufSubstrAfter(&os->variant, strlen("Microsoft Windows ") - 1); // 11 家庭中文版
+        ffStrbufSubstrAfter(&os->variant, strlen("Microsoft Windows ") - 1); // Server 2022 Datacenter / 11 家庭中文版
+
+        if(ffStrbufStartsWithS(&os->variant, "Server "))
+        {
+            ffStrbufAppendS(&os->name, " Server");
+            ffStrbufAppendS(&os->prettyName, " Server");
+            ffStrbufSubstrAfter(&os->variant, strlen(" Server") - 1);
+        } // 2022 Datacenter / 11 家庭中文版
```
Comment (LinusDierheimer) [inline src/detection/os/os_windows.cpp:49]: Can the please keep everything in english?

Comment (CarterLi) [reply src/detection/os/os_windows.cpp]: It was actually the Windows version printed by OS. I left it here to make the string operations easier to understand.

I removed the comments.

**Comment by CarterLi** [comment]
> CI seems fine, and i can't really judge the windows code. Can i merge this now or are you still working on it?

That's it for now.

We need volunteers to test it, especially on different Windows versions ( 10, 8 and 7 maybe ), as I have only Windows 11 machine.

**Comment by LinusDierheimer** [comment]
I guess they will automatically come, as soon as it gets released. Like you when i released a very bare bones version of MacOS support.

**Pull request merged**